### ISSUE TITLE
Change FIGS predict_proba to use sigmoid?

### ISSUE THREAD

**Issue by mepland** [csinva/imodels#149]
We should use a [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function), like `scipy.special.expit`, to constrain the FIGS classifier's `predict_proba` between 0 and 1, instead of `np.clip(preds, a_min=0., a_max=1.)`. This matches what XGBoost does, see `binary:logistic` [here](https://xgboost.readthedocs.io/en/stable/parameter.html#learning-task-parameters), and is nicer mathematically as it is a continuous function. If changed, we should also make a note of this in the documentation, and any future papers.

Users can further [calibrate](https://scikit-learn.org/stable/modules/calibration.html) the output probabilities with `CalibratedClassifierCV`, or some other method, if they desire.
